pkg/storage/mto_milestone.go

```diff
@@ -88,3 +88,39 @@ func MTOMilestoneGetByID(np sqlutils.NamedPreparer, _ *zap.Logger, id uuid.UUID)
 
 	return returned, nil
 }
+
+// MTOMilestoneSolutionLinkCreate creates a new MTO Milestone-Solution link in the database
+func MTOMilestoneSolutionLinkCreate(
+	np sqlutils.NamedPreparer,
+	_ *zap.Logger,
+	mtoMilestoneSolutionLink *models.MTOMilestoneSolutionLink,
+) (*models.MTOMilestoneSolutionLink, error) {
+
+	if mtoMilestoneSolutionLink.ID == uuid.Nil {
+		mtoMilestoneSolutionLink.ID = uuid.New()
+	}
+
+	link, procErr := sqlutils.GetProcedure[models.MTOMilestoneSolutionLink](np, sqlqueries.MTOMilestone.CreateMilestoneSolutionLink, mtoMilestoneSolutionLink)
+	if procErr != nil {
+		return nil, fmt.Errorf("issue creating new MTO Milestone-Solution link: %w", procErr)
+	}
+
+	return link, nil
+}
+
+// MTOMilestoneSolutionLinkGetByMilestoneID returns all MTO Milestone-Solution links for a given Milestone ID
+func MTOMilestoneSolutionLinkGetByMilestoneID(
+	np sqlutils.NamedPreparer,
+	_ *zap.Logger,
+	milestoneID uuid.UUID,
+) ([]*models.MTOMilestoneSolutionLink, error) {
+
+	arg := map[string]interface{}{"milestone_id": milestoneID}
+
+	returned, procErr := sqlutils.SelectProcedure[models.MTOMilestoneSolutionLink](np, sqlqueries.MTOMilestone.GetMilestoneSolutionLinksByMilestoneID, arg)
```

Comment:
I'm not sure that you need to fetch links explicitly here, I think you can validate by fetching linked solutions instead. We don't really need to fetch links, just the solutions / milestones themselves.

Reply:
You're right. I thought about this quite a bit yesterday. In an ideal world our store would be mocked and we could do flow assertions, but the only way to validate the linking table is functioning as expected was to create a simple fetch. I feel like for unit testing / validation purposes having a fetch method is worthwhile.